crates/provider/Cargo.toml

```
@@ -6,3 +6,10 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
+async-trait = "0.1.68"
+futures = "0.3.28"
+napi = { version="2.12.7", features=["async"]}
+napi-derive = "2.12.5"
+serde = {verson="1.0.163", features = ["derive"]}
+serde_json = "1.0.96"
+tokio = { version = "1.28.2", features = ["process", "macros", "fs", "time"] }
```

What we could do is [hoist the dependencies](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-dependencies-table) to the ```workspace's Cargo.toml```, and additionaly only put the version without the minor so the patches are applied automatically at each build:

```rust
[workspace]
members = [
  "crates/support",
  "crates/configuration",
  "crates/orchestrator",
  "crates/provider",
  "crates/test-runner"
]

[workspace.dependencies]
async-trait = "0.1"
futures = "0.3"
napi = "2.12"
napi-derive = "2.12"
serde = "1.0"
serde_json = "1.0"
tokio = "1.28"
```

And then in the ```crate's Cargo.toml``` you can do this:
```suggestion
async-trait = { workspace = true }
futures = { workspace = true }
napi = { workspace = true, features=["async"]}
napi-derive = { workspace = true }
serde = {workspace = true, features = ["derive"]}
serde_json = { workspace = true }
tokio = { workspace = true, features = ["process", "macros", "fs", "time"] }
```

this way, all crates share the same version of dependencies but can get what they want from it.